packages/server/test/client.spec.ts

```diff
@@ -82,14 +83,73 @@ const MOCK_PROVIDER: Provider = {
 };
 
 describe('OpenFeatureClient', () => {
-  beforeAll(() => {
+  beforeEach(() => {
     OpenFeature.setProvider(MOCK_PROVIDER);
   });
 
   afterEach(() => {
     jest.clearAllMocks();
   });
 
+  describe('Requirement 1.1.8', () => {
+    class mockAsyncProvider implements Provider {
```

Comment:
I think we have a testing util provider we can use for this, but up to you.

Reply:
I didn't notice one that would work for this specific use case. However, I definitely think we could consolidate the mock providers in a future PR.